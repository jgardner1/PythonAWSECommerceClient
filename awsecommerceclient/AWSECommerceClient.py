from boto.connection import AWSQueryConnection
from Parser import Parser
import time
import urllib

class AWSECommerceClient(object):

    def __init__(self,
            aws_access_key_id,
            aws_associate_tag,
            aws_secret_access_key,
    ):
        self.aws_access_key_id = aws_access_key_id
        self.aws_associate_tag = aws_associate_tag
        self.aws_secret_access_key = aws_secret_access_key

    def method(self, operation, **kwargs):
        aws_conn = AWSQueryConnection(
            aws_access_key_id=self.aws_access_key_id,
            aws_secret_access_key=self.aws_secret_access_key, is_secure=False,
            host='ecs.amazonaws.com')

        aws_conn.SignatureVersion = '2'
        kwargs.update(dict(
            Service='AWSECommerceService',
            Version=Parser.version,
            SignatureVersion=aws_conn.SignatureVersion,
            AWSAccessKeyId=self.aws_access_key_id,
            AssociateTag=self.aws_associate_tag,
            Timestamp=time.strftime("%Y-%m-%dT%H:%M:%S", time.gmtime()),
            Operation=operation,
        ))
        verb = 'GET'
        path = '/onca/xml'
        qs, signature = aws_conn.get_signature(kwargs, verb, path)
        qs = path + '?' + qs + '&Signature=' + urllib.quote(signature)

        return Parser.parse_file(aws_conn._mexe(verb, qs, None, headers={}))

    def itemSearch(self, **kwargs): 
        return self.method('ItemSearch', **kwargs)

