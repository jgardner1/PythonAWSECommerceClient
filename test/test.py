#!/usr/bin/env python
from awsecommerceclient import AWSECommerceClient

client = AWSECommerceClient(
    '1KHB4V2W5HGHCAHD2M82',
    'onlyeee-20',
    'wx+WYK5aXWBJQmlut0886531i3THLP0CY2SRfxrX')

response = client.ItemLookup(
    ItemId='B000H6SXMY',
    ResponseGroup='Reviews')

from pprint import pprint
pprint(response['Items'])
