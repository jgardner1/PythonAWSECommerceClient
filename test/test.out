{u'Items': [{u'Item': [{u'ASIN': u'B000H6SXMY',
                        u'CustomerReviews': {u'AverageRating': Decimal('4.5'),
                                             u'Review': [{u'ASIN': u'B000H6SXMY',
                                                          u'Content': u"All these years later, the show still works.  It's the kind of show that was made possible and unique by a fantastic ensemble.  The writing never needed to be even great for a show to make you laugh.  It was by and large an opportunity for a perfectly cast group of friends to be brought to life by some talented actors.  While these actors may never go on to do anything else that comes close to what they did with friends, the show will still be remembered for a long time.",
                                                          u'CustomerId': u'A13JKDAAQP9D58',
                                                          u'Date': u'2009-10-01',
                                                          u'HelpfulVotes': 0,
                                                          u'Rating': Decimal('5'),
                                                          u'Reviewer': {u'CustomerId': u'A13JKDAAQP9D58',
                                                                        u'Location': u'CA, USA',
                                                                        u'Name': u'Michael A. Reilly'},
                                                          u'Summary': u'Great Easy Viewing',
                                                          u'TotalVotes': 0},
                                                         {u'ASIN': u'B000H6SXMY',
                                                          u'Content': u"I've been a Friends fan for years and have watched the episodes over and over and everytime they still make me laugh out loud. I would reccommend buying this set for anyone who is a fan of the show, its definately worth it!\n<br />\n<br />The packaging is really neat, and the book that comes with it has some photos of the cast and snap shots of a few of the episodes. Its also has a brief description of every episode.",
                                                          u'CustomerId': u'AWB2NXI7GDRJL',
                                                          u'Date': u'2009-09-30',
                                                          u'HelpfulVotes': 0,
                                                          u'Rating': Decimal('5'),
                                                          u'Reviewer': {u'CustomerId': u'AWB2NXI7GDRJL',
                                                                        u'Name': u'C. M. Perella'},
                                                          u'Summary': u'A Must Own!!',
                                                          u'TotalVotes': 0},
                                                         {u'ASIN': u'B000H6SXMY',
                                                          u'Content': u'Best show I have Ever purchased in my life I watch friends just about ever night.',
                                                          u'CustomerId': u'A2ZE3MY1Q59ZLF',
                                                          u'Date': u'2009-09-30',
                                                          u'HelpfulVotes': 0,
                                                          u'Rating': Decimal('5'),
                                                          u'Reviewer': {u'CustomerId': u'A2ZE3MY1Q59ZLF',
                                                                        u'Location': u'Wisconsin',
                                                                        u'Name': u'J.Rich',
                                                                        u'Nickname': u'J.Rich'},
                                                          u'Summary': u'AMAZING',
                                                          u'TotalVotes': 0},
                                                         {u'ASIN': u'B000H6SXMY',
                                                          u'Content': u"If i had received this product i would be able to review this product but of course i cant because i didn't.",
                                                          u'CustomerId': u'A3LKB9L8S70B5M',
                                                          u'Date': u'2009-09-25',
                                                          u'HelpfulVotes': 0,
                                                          u'Rating': Decimal('1'),
                                                          u'Reviewer': {u'CustomerId': u'A3LKB9L8S70B5M',
                                                                        u'Name': u'Stephen Baker'},
                                                          u'Summary': u'I would if i could.',
                                                          u'TotalVotes': 7},
                                                         {u'ASIN': u'B000H6SXMY',
                                                          u'Content': u'The transaction was good, the box of the collection was a little messed up, the corners bend and all but it was good. ',
                                                          u'CustomerId': u'A1ON89LB8P8GX5',
                                                          u'Date': u'2009-09-22',
                                                          u'HelpfulVotes': 0,
                                                          u'Rating': Decimal('4'),
                                                          u'Reviewer': {u'CustomerId': u'A1ON89LB8P8GX5',
                                                                        u'Name': u'F. Polo'},
                                                          u'Summary': u'Good but not great shape',
                                                          u'TotalVotes': 1}],
                                             u'TotalReviewPages': 52,
                                             u'TotalReviews': 257}}],
             u'Request': {u'IsValid': u'True',
                          u'ItemLookupRequest': {u'Condition': None,
                                                 u'DeliveryMethod': None,
                                                 u'IdType': None,
                                                 u'ItemId': [u'B000H6SXMY'],
                                                 u'MerchantId': u'Amazon',
                                                 u'OfferPage': 1,
                                                 u'ResponseGroup': [u'Reviews'],
                                                 u'ReviewPage': 1,
                                                 u'ReviewSort': u'-SubmissionDate',
                                                 u'VariationPage': None}}}],
 u'OperationRequest': {u'Arguments': {u'Argument': [{u'Name': u'Operation',
                                                     u'Value': u'ItemLookup'},
                                                    {u'Name': u'Service',
                                                     u'Value': u'AWSECommerceService'},
                                                    {u'Name': u'ItemId',
                                                     u'Value': u'B000H6SXMY'},
                                                    {u'Name': u'SignatureMethod',
                                                     u'Value': u'HmacSHA256'},
                                                    {u'Name': u'AWSAccessKeyId',
                                                     u'Value': u'1KHB4V2W5HGHCAHD2M82'},
                                                    {u'Name': u'Timestamp',
                                                     u'Value': u'2009-10-08T23:28:37'},
                                                    {u'Name': u'Signature',
                                                     u'Value': u'I/GngfuabIfNbqlt5gwo+1aNZNBJmHwtvtKOHKLYGfc='},
                                                    {u'Name': u'ResponseGroup',
                                                     u'Value': u'Reviews'},
                                                    {u'Name': u'AssociateTag',
                                                     u'Value': u'onlyeee-20'},
                                                    {u'Name': u'SignatureVersion',
                                                     u'Value': u'2'},
                                                    {u'Name': u'Version',
                                                     u'Value': u'2009-10-01'}]},
                       u'RequestId': u'd1f41fe0-397b-4651-9ab2-ef239d1647f6',
                       u'RequestProcessingTime': 0.041355999999999997}}
